ingestion/src/metadata/utils/importer.py

```diff
@@ -43,6 +43,32 @@ class DynamicImportException(Exception):
     Raise it when having issues dynamically importing objects
     """
 
+    def __init__(self, module: str, key: str = None, cause: Exception = None):
```

Comment:
nit. can we rename `cause` to `err` or `err_message` or `message`?

Reply:
why? cause seems to be a more appropriate name since its the error that caused this one.

Reply:
should we use the StackTraceError instead? Since it's our "common" way of handling errors

Reply:
Going to push back on this one. If any of our dynamic imports fail I'd rather have the python exception as raw as possible so we can investigate. If we start processing it in any way we might lose important information.
I think a dynamic import failure is "catastrophic" in the sense that its not an exception we should get in any execution path so if we do meet it it should have all the context ready for inspection.
This is unlike missing entities or failed API calls which can fail more gracefully.